src/RoomInfoListItem.cpp

```diff
@@ -324,6 +324,20 @@ RoomInfoListItem::updateUnreadMessageCount(int count, int highlightedCount)
         update();
 }
 
+unsigned short int
+RoomInfoListItem::calculateImportance() const
+{
+        // 0: All messages and minor events read
+        // 1: Contains unread minor events (joins/notices/muted messages)
+        // 2: Contains unread messages
+        // 3: Contains mentions
+        // 4: Is a room invite
+        return (hasUnreadMessages_) +
```

Comment:
While this works, I think it may be better to be explicit here and do an if else chain. I would also prefer it, if you could make this an enum instead of a calculation, to improve readability. I.e.:

```
enum NotificationImportance : unsigned short {
AllEventsRead = 0,
SilencedUnreadEvents = 1,
UnreadEvents = 2,
Mentions = 3,
Invite = 4
};

if (isInvite())
 return Invite;
if (unreadHighlightMsgCount_)
 return Mentions;
if (unreadMsgCount_)
 return ...
```

Although you may want to tweak the names.